test/Test_Menhir.re

```diff
@@ -0,0 +1,194 @@
+open Haz3lmenhir;
+open Alcotest;
+open Haz3lcore;
+
+let testable_ast = testable(Fmt.using(AST.show_exp, Fmt.string), (==));
+
+let exp_typ =
+  testable(
+    Fmt.using(Haz3lcore.Exp.show, Fmt.string),
+    Haz3lcore.DHExp.fast_equal,
+  );
+
+let alco_check = exp_typ |> Alcotest.check;
+
+// Existing recovering parser
+let make_term_parse = (s: string) =>
+  MakeTerm.from_zip_for_sem(Option.get(Printer.zipper_of_string(s))).term;
+
+let parser_test = (name: string, exp: Term.Exp.t, actual: string) =>
+  test_case(
+    name,
+    `Quick,
+    () => {
+      alco_check("Does not match MakeTerm", exp, make_term_parse(actual));
+
+      alco_check(
+        name ++ " matches expected type",
+        exp,
+        Haz3lmenhir.Conversion.Exp.of_menhir_ast(
+          Haz3lmenhir.Interface.parse_program(actual),
+        ),
+      );
+    },
+  );
```

Comment:
@green726 I added a few tests of the parser for different forms below. These are asserting that it matches the expected value and that it's equivalent to what the tylr version of the parser does. There's a few failing cases right now:

1. Data constructors seem to be failing
2. My named_fun test is failing (I haven't verified that the test isn't wrong)
3. There's some issue with the nested integer operators.


The tests aren't conclusive. I'll try to add some more soon but I feel like we're close to covering all the frequently used cases.

Reply:
I've been working on adding some tests as well (sorry, I believe that I forgot to push them). I can cross-check our two results for the named functions. Yeah, it does seem that we are approaching that.